File: python/tests/transport/spi_test_linux.py
import os
import sys

from periph.transport.spi_linux import SPITransport

SPI_BUS    = int(os.environ.get('LINUX_SPI_BUS',    '0'))
SPI_DEVICE = int(os.environ.get('LINUX_SPI_DEVICE', '0'))
SPI_MODE   = int(os.environ.get('LINUX_SPI_MODE',   '0'))
SPI_SPEED  = int(os.environ.get('LINUX_SPI_SPEED',  '1000000'))

passed = 0
failed = 0


def check_true(label, condition):
    global passed, failed
    if condition:
        print('PASS', label)
        passed += 1
    else:
        print('FAIL', label)
        failed += 1


transport = SPITransport(SPI_BUS, SPI_DEVICE, mode=SPI_MODE, max_speed_hz=SPI_SPEED)

transport.write(bytes([0x00]))
check_true('write accepted', True)

data = transport.read(1)
check_true('read returns 1 byte', len(data) == 1)

data = transport.write_read(bytes([0x00]), 1)
check_true('write_read returns 1 byte', len(data) == 1)

transport.close()

print('===DONE: {} passed, {} failed==='.format(passed, failed))
sys.exit(0 if failed == 0 else 1)

File: cpp/tests/transport/spi_test_linux/spi_test_linux.cpp
#include <cstdio>
#include "SPITransportLinux.h"

#ifndef TEST_SPI_BUS
#define TEST_SPI_BUS    0
#endif
#ifndef TEST_SPI_DEVICE
#define TEST_SPI_DEVICE 0
#endif
#ifndef TEST_SPI_MODE
#define TEST_SPI_MODE   0
#endif
#ifndef TEST_SPI_SPEED
#define TEST_SPI_SPEED  1000000
#endif

static int passed = 0;
static int failed = 0;

static void check_true(const char* label, bool condition) {
    if (condition) { printf("PASS %s\n", label); passed++; }
    else           { printf("FAIL %s\n", label); failed++; }
}

int main() {
    SPITransportLinux transport(TEST_SPI_BUS, TEST_SPI_DEVICE, TEST_SPI_MODE, TEST_SPI_SPEED);

    uint8_t cmd[1] = {0x00};
    transport.write(cmd, 1);
    check_true("write accepted", true);

    uint8_t buf[1] = {0};
    transport.read(buf, 1);
    check_true("read returns data", true);

    transport.write_read(cmd, 1, buf, 1);
    check_true("write_read returns data", true);

    printf("===DONE: %d passed, %d failed===\n", passed, failed);
    return failed == 0 ? 0 : 1;
}
